src/codegen/codegen_ispc_visitor.cpp

```diff
@@ -683,43 +551,84 @@ void CodegenIspcVisitor::print_backend_compute_routine_decl() {
     }
 }
 
-void CodegenIspcVisitor::determine_target() {
+bool CodegenIspcVisitor::check_incompatibilities() {
     const auto& has_incompatible_nodes = [this](const ast::Ast& node) {
         return !collect_nodes(node, incompatible_node_types).empty();
     };
 
+    const auto get_name_from_symbolTypeVector = [](const SymbolType& var) -> const std::string& {
+        return var->get_name();
+    };
+
+    // instance vars
+    if (check_incompatible_var_name<SymbolType>(codegen_float_variables,
+                                                get_name_from_symbolTypeVector) ||
+        check_incompatible_var_name<SymbolType>(codegen_shadow_variables,
+                                                get_name_from_symbolTypeVector) ||
+        check_incompatible_var_name<IndexVariableInfo>(codegen_int_variables,
+                                                       [](const IndexVariableInfo& var)
+                                                           -> const std::string& {
+                                                           return var.symbol->get_name();
+                                                       }) ||
+
+
+        check_incompatible_var_name<std::string>(info.currents,
+                                                 [](const std::string& var) -> const std::string& {
+                                                     return var;
+                                                 }) ||
+
+
+        // global vars
+        // info.top_local_variables is not checked because it should be addressed by the
+        // renameIspcVisitor
+        check_incompatible_var_name<SymbolType>(info.global_variables,
+                                                get_name_from_symbolTypeVector) ||
+
+
+        check_incompatible_var_name<SymbolType>(info.constant_variables,
+                                                get_name_from_symbolTypeVector)) {
+        return true;
+    }
+
+    // ion vars
+    for (const auto& ion: info.ions) {
+        if (check_incompatible_var_name<std::string>(
+                ion.writes, [](const std::string& var) -> const std::string& { return var; })) {
+            return true;
+        }
+    }
+
+
+    emit_fallback = std::vector<bool>(BlockType::BlockTypeEnd, false);
+
     if (info.initial_node) {
         emit_fallback[BlockType::Initial] =
-            has_incompatible_nodes(*info.initial_node) ||
+            emit_fallback[BlockType::Initial] || has_incompatible_nodes(*info.initial_node) ||
             visitor::calls_function(*info.initial_node, "net_send") || info.require_wrote_conc;
     } else {
-        emit_fallback[BlockType::Initial] = info.net_receive_initial_node ||
+        emit_fallback[BlockType::Initial] = emit_fallback[BlockType::Initial] ||
```

Comment:
> `emit_fallback[BlockType::Initial] = emit_fallback[BlockType::Initial] ||` 

I might be missing something here: why do we `||` with `emit_fallback[x]` itself ?

Reply:
so that it does not override a 0 if we had a 1 before. In this way it becomes more modular and scalable